src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,617 @@
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request_5[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 6
+        # TH sends the Read Request Message to the DUT to read a global attribute from all clusters at all Endpoints
+        # AttributePath = [[Attribute = Specific Global Attribute]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(6, "Send Request Message to read one global attribute from all clusters on all endpoints")
+        # endpoint = [Clusters.Objects.Descriptor.Attributes.AttributeList]
+        read_request_6 = await self.default_controller.ReadAttribute(self.dut_node_id, [attribute_list])
+        returned_endpoints = read_request_6[0].keys()
+        for i in range(3):
+            returned_endpoints = read_request_6[i].keys()
+            # Check if chip.clusters.Objects.Descriptor is in output
+            asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request_6[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 7
+        # TH sends the Read Request Message to the DUT to read all attributes from a cluster at all Endpoints
+        # AttributePath = [[Cluster = Specific ClusterID]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(7, "Send Request Message to read all attributes from one cluster at all endpoints")
+        # endpoint = [Clusters.Objects.Descriptor]
+        read_request_7 = await self.default_controller.ReadAttribute(self.dut_node_id, [descriptor_obj])
+        returned_endpoints = read_request_7[0].keys()
+        for i in range(3):
+            returned_endpoints = read_request_7[i].keys()
+            # Check if chip.clusters.Objects.Descriptor is in output
+            asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request_7[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 8
+        # TH sends the Read Request Message to the DUT to read all attributes from all clusters at one Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(8, "Send Request Message to read all attributes from all clusters at one endpoint")
+        read_request_8 = await self.default_controller.ReadAttribute(self.dut_node_id, [0])
+        returned_endpoints = read_request_8[0].keys()
```

Comment:
Same comments as above - check that only the requested endpoint is returned, exact match on expected clusters.